run/opencl/opencl_aes_plain.h

```diff
@@ -80,23 +87,29 @@ typedef struct aes_key_st {
 /**
  * Copy tables to local memory
  */
-INLINE void aes_table_init(__local aes_local_t *lt)
+INLINE void aes_enc_table_init(__local aes_local_t *lt)
 {
 	for (uint i = THREAD; i < 256; i += LWS) {
 		lt->Te0[i] = Te0[i];
 		lt->Te1[i] = Te1[i];
 		lt->Te2[i] = Te2[i];
 		lt->Te3[i] = Te3[i];
+		if (i < 10)
+			lt->rcon[i] = rcon[i];
+	}
+	barrier(CLK_LOCAL_MEM_FENCE);
+}
+
+INLINE void aes_dec_table_init(__local aes_local_t *lt)
+{
+	for (uint i = THREAD; i < 256; i += LWS) {
 		lt->Td0[i] = Td0[i];
 		lt->Td1[i] = Td1[i];
 		lt->Td2[i] = Td2[i];
 		lt->Td3[i] = Td3[i];
 		lt->Td4[i] = Td4[i];
-		if (i < 10)
-			lt->rcon[i] = rcon[i];
 	}
-
-	barrier(CLK_LOCAL_MEM_FENCE);
+	/* Intentionally no barrier here - it will happen later */
```

Comment:
Maybe explain (in the comment) when the barrier "will happen later". I don't see it from skimming the code.